src/group_theory/divisible.lean

```diff
@@ -0,0 +1,255 @@
+  div_zero := λ a, dif_pos rfl,
+  div_cancel := λ n a hn, begin
+    rw [dif_neg hn],
+    generalize_proofs h1,
+    exact h1.some_spec.2,
+  end }
+
+end add_comm_group
+
+@[priority 100]
+instance divisible_by_int_of_char_zero {𝕜} [division_ring 𝕜] [char_zero 𝕜] : divisible_by 𝕜 ℤ :=
+{ div := λ q n, q / n,
+  div_zero := λ q, by norm_num,
+  div_cancel := λ n q hn,
+    by rw [zsmul_eq_mul, (int.cast_commute n _).eq, div_mul_cancel q (int.cast_ne_zero.mpr hn)] }
+
+namespace group
+
+variables (A : Type*) [group A]
+
+/--
+A group is `ℤ`-rootable if it is `ℕ`-rootable.
+-/
+@[to_additive add_group.divisible_by_int_of_divisible_by_nat
+"An additive group is `ℤ`-divisible if it is `ℕ`-divisible."]
+def rootable_by_int_of_rootable_by_nat [rootable_by A ℕ] : rootable_by A ℤ :=
+{ root := λ a z, match z with
+  | (n : ℕ) := rootable_by.root a n
+  | -[1+n] := (rootable_by.root a (n + 1))⁻¹
+  end,
+  root_zero := λ a, rootable_by.root_zero a,
+  root_cancel := λ n a hn, begin
+    induction n,
+    { change (rootable_by.root a _) ^ _ = a,
+      norm_num,
+      rw [rootable_by.root_cancel],
+      rw [int.of_nat_eq_coe] at hn,
+      exact_mod_cast hn, },
+    { change ((rootable_by.root a _) ⁻¹)^_ = a,
+      norm_num,
+      rw [rootable_by.root_cancel],
+      norm_num, }
+  end}
+
+/--A group is `ℕ`-rootable if it is `ℤ`-rootable
+-/
+@[to_additive add_group.divisible_by_nat_of_divisible_by_int
+"An additive group is `ℕ`-divisible if it `ℤ`-divisible."]
+def rootable_by_nat_of_rootable_by_int [rootable_by A ℤ] : rootable_by A ℕ :=
+{ root := λ a n, rootable_by.root a (n : ℤ),
+  root_zero := λ a, rootable_by.root_zero a,
+  root_cancel := λ n a hn, begin
+    have := rootable_by.root_cancel a (show (n : ℤ) ≠ 0, by exact_mod_cast hn),
+    norm_num at this,
+    exact this,
+  end }
+
+end group
+
+section hom
+
+variables {α A B : Type*}
+variables [has_zero α] [monoid A] [monoid B] [has_pow A α] [has_pow B α] [rootable_by A α]
+variables (f : A → B)
+
+/--
+If `f : A → B` is a surjective homomorphism and `A` is `α`-rootable, then `B` is also `α`-rootable.
+-/
+@[to_additive "If `f : A → B` is a surjective homomorphism and
+`A` is `α`-divisible, then `B` is also `α`-divisible."]
+noncomputable def function.surjective.rootable_by (hf : function.surjective f)
+  (hpow : ∀ (a : A) (n : α), f (a ^ n) = f a ^ n) : rootable_by B α :=
+rootable_by_of_pow_left_surj _ _ $ λ n hn x,
+  let ⟨y, hy⟩ := hf x in ⟨f $ rootable_by.root y n, (by rw [←hpow (rootable_by.root y n) n,
+    rootable_by.root_cancel _ hn, hy] : _ ^ _ = x)⟩
+
+end hom
+
+section quotient
+
+variables (α : Type*) {A : Type*} [comm_group A] (B : subgroup A)
+
+/-- Any quotient group of a rootable group is rootable. -/
+@[to_additive quotient_add_group.divisible_by
+"Any quotient group of a divisible group is divisible"]
+noncomputable def quotient_group.rootable_by [rootable_by A ℕ] : rootable_by (A ⧸ B) ℕ :=
+quotient_group.mk_surjective.rootable_by _ $ λ _ _, rfl
```

Comment:
Did you deliberately not make this an instance?

Reply:
That is a mistake